runtime/gc_include/ObjectAllocationAPI.hpp

```diff
@@ -394,12 +399,22 @@ class MM_ObjectAllocationAPI
 		if (J9VMTHREAD_COMPRESS_OBJECT_REFERENCES(currentThread)) {
 			J9ObjectCompressed *objectHeader = (J9ObjectCompressed*) instance;
 			objectHeader->clazz = (uint32_t)(uintptr_t)clazz;
+			if (ensureHashed) {
+				VM_AtomicSupport::bitOrU32(
+						(uint32_t*)&objectHeader->clazz,
+						(uint32_t)OBJECT_HEADER_HAS_BEEN_HASHED_IN_CLASS);
+			}
```

Comment:
I am not sure there is need to set flag atomically. Inline allocation uses TLH so nobody should interfere

Reply:
Also, please don't write to the class twice - OR the bit into the local before assigning to object->clazz.

Reply:
Is the current change fine? Or did you want a new local

Reply:
Please just modify the `clazz` variable and perform the write once.

Reply:
I assume you're looking for
```cpp
if (J9_ARE_ANY_BITS_SET(clazz->classFlags, J9ClassEnsureHashed)) {
	clazz |= (uintptr_t)OBJECT_HEADER_HAS_BEEN_HASHED_IN_CLASS;
}
objectHeader->clazz = (uint32_t)(uintptr_t)clazz;
```
Wouldn't modifying `clazz` impact the 
```cpp
j9objectmonitor_t initialLockword = VM_ObjectMonitor::getInitialLockword(currentThread->javaVM, clazz);
```
?

Reply:
Yes, you're right - let's leave it the way it is now.